zk-gadget/src/tests.rs

```diff
@@ -0,0 +1,9 @@
+use crate::mock::*;
+
+#[test]
+fn gadget_starts() {
+    let (ext, runtime) = new_test_ext();
+    ext.lock().execute_with(|| {
```

Comment:
Note that `lock()` is used elsewhere in the code *while* this lock for the test is running. Since execute_with requires `&mut self`, an `RwLock` wouldn't work either. I don't think this pattern will work. We will need to think of something else.